apps/verticalhome/js/pin_app_navigation.js

```diff
@@ -0,0 +1,294 @@
+'use strict';
+/* global ApplicationSource */
+/* global BookmarkSource */
+/* global CollectionSource */
+/* global configurator */
+/* global dispatchEvent */
+/* global GaiaGrid */
+
+(function(exports) {
+
+
+  /**
+   * This is PinAppNavigation's constructor.
+   * @param {Object} list [DOM container that contains pin elements]
+   */
+  function PinAppNavigation(list) {
+    if (!list && typeof list !== 'object') {
+      console.log('List of pin apps must be an object');
+      return;
+    }
+
+    this.elemList = list;
+    window.removeEventListener('keydown', this);
```

Comment:
The this object was just created. It is impossible that it is already registered as an event handler. So no need to remove it.